feeder/src/vote.ts

```diff
@@ -295,33 +295,69 @@ interface VoteArgs {
 }
 
 export async function vote(args: VoteArgs): Promise<void> {
-  const client = new LCDClient({
-    URL: args.lcdAddress,
+  const maxLCDIndex = args.lcdAddress.length - 1
+  let currentLCDIndex = 0
+
+  let client = new LCDClient({
+    URL: args.lcdAddress[currentLCDIndex],
     chainID: args.chainID,
     gasPrices: args.gasPrices,
   })
   const rawKey: RawKey = await initKey(args.keyPath, args.password)
   const valAddrs = args.validator || [rawKey.valAddress]
   const voterAddr = rawKey.accAddress
-  const wallet = new Wallet(client, rawKey)
+  let wallet = new Wallet(client, rawKey)
+  let switchLCD = false
 
   while (true) {
     const startTime = Date.now()
 
-    await processVote(client, wallet, args.source, valAddrs, voterAddr).catch((err) => {
-      if (err.isAxiosError && err.response) {
-        console.error(err.message, err.response.data)
-      } else {
-        console.error(err.message)
+    if (switchLCD) {
+      switchLCD = false
+      if (++currentLCDIndex > maxLCDIndex) {
+        currentLCDIndex = 0
       }
+      client = new LCDClient({
+        URL: args.lcdAddress[currentLCDIndex],
+        chainID: args.chainID,
+        gasPrices: args.gasPrices,
+      })
+      wallet = new Wallet(client, rawKey)
+      console.log(
+        'Switched to LCD address ' + currentLCDIndex + '(' + args.lcdAddress[currentLCDIndex] + ')'
+      )
+    }
```

Comment:
I'd suggest putting the LCD switching logic into its own separate function `switchLCD`, which then returns a new LCDClient. In that case, you'd just need to put in the function call in the catch clause below. Makes things more readable. Other than that, looks good to me 👍🏼

Reply:
I moved the lcd switch logic to its own function